src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -279,7 +280,9 @@ ASTPtr PostgreSQLReplicationHandler::getCreateNestedTableQuery(StorageMaterializ
     auto table_structure = std::make_unique<PostgreSQLTableStructure>(fetchPostgreSQLTableStructure(tx, table_name, postgres_schema, true, true, true));
     if (!table_structure)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Failed to get PostgreSQL table structure");
-    return storage->getCreateNestedTableQuery(std::move(table_structure));
+
+    auto table_override = ASTTableOverride::tryGetTableOverride(current_database_name, table_name);
+    return storage->getCreateNestedTableQuery(std::move(table_structure), table_override->as<ASTTableOverride>());
```

Comment:
Seems like `table_override` is `nullptr` if `CREATE DATABASE` query does not contain `TABLE OVERRIDE` clauses.